File: app/src/main/java/io/github/zwieback/familyfinance/business/operation/lifecycle/creator/OperationCreator.java
package io.github.zwieback.familyfinance.business.operation.lifecycle.creator;

import android.content.Context;

import java.util.TreeSet;

import io.github.zwieback.familyfinance.core.lifecycle.creator.EntityCreator;
import io.github.zwieback.familyfinance.core.model.Operation;
import io.requery.Persistable;
import io.requery.reactivex.ReactiveEntityStore;

public class OperationCreator extends EntityCreator<Operation> {

    public OperationCreator(Context context, ReactiveEntityStore<Persistable> data) {
        super(context, data);
    }

    @Override
    protected Iterable<Operation> buildEntities() {
        return new TreeSet<>(this);
    }

    @Override
    public int compare(Operation left, Operation right) {
        return left.getDate().compareTo(right.getDate());
    }
}

File: app/src/main/java/io/github/zwieback/familyfinance/business/operation/lifecycle/creator/OperationCreator.kt
package io.github.zwieback.familyfinance.business.operation.lifecycle.creator

import android.content.Context
import io.github.zwieback.familyfinance.core.lifecycle.creator.EntityCreator
import io.github.zwieback.familyfinance.core.model.Operation
import io.requery.Persistable
import io.requery.reactivex.ReactiveEntityStore

class OperationCreator(
    context: Context,
    data: ReactiveEntityStore<Persistable>
) : EntityCreator<Operation>(context, data) {

    override fun buildEntities(): Iterable<Operation> {
        return sortedSetOf(this)
    }

    override fun compare(left: Operation, right: Operation): Int {
        return left.date.compareTo(right.date)
    }
}
